src/front/glsl/builtins.rs

```diff
@@ -165,184 +216,84 @@ pub fn inject_builtin(declaration: &mut FunctionDeclaration, module: &mut Module
                     class,
                 };
 
-                let vector = match (dim, shadow) {
-                    (0b00, false) => TypeInner::Scalar {
-                        kind: Sk::Float,
-                        width,
-                    },
-                    (0b00, true) | (0b11, _) => TypeInner::Vector {
-                        size: VectorSize::Tri,
-                        kind: Sk::Float,
-                        width,
-                    },
-                    (_, _) => {
-                        let size = match dim + arrayed as u32 + shadow as u32 {
-                            1 => VectorSize::Bi,
-                            2 => VectorSize::Tri,
-                            _ => VectorSize::Quad,
-                        };
-
-                        TypeInner::Vector {
-                            size,
-                            kind: Sk::Float,
-                            width,
-                        }
-                    }
+                let num_coords_from_dim = match dim {
+                    0b00 => 1, // 1D
+                    0b01 => 2, // 2D
+                    0b10 => 3, // Cube
+                    _ => 3, // 3D
                 };
+                let mut num_coords = num_coords_from_dim;
 
-                let mut args = vec![image, vector];
+                if shadow {
+                    num_coords += 1;
```

Comment:
You can make these as `num_coords += shadow as u8`

Reply:
This is now changed.